package.json

```diff
@@ -28,6 +28,7 @@
     "rollup-plugin-css-only": "^3.1.0",
     "rollup-plugin-svg": "^2.0.0",
     "semver": "^7.3.5",
+    "shelljs": "^0.8.5",
     "shx": "^0.3.4",
```

Comment:
Kinda too lazy to look this up - but do we really need _both_ shelljs and shx? Looks like the shelljs package itself is only being used in a single package, i.e. polyExplorer. Definitely not a hot candidate for hoisting if that's the case :D

Reply:
Well, it's now first-level since it's used in the root (indirectly) and also polyExplorer and bundle (indirectly). We can implicitly assume that the second dependence will be re-used from the first, but anyway it's better to hoist so that we can fix the version and not assume that shelljs will be installed with shx. It's mostly a judgment call, but explicit is better than implicit, and this actually fixes the issue, so...

Reply:
You mean because it's part of the polyExplorer build? Still not a reason to me to hoist it - before polyExplorer is built, its dependencies are being installed... Am I missing something? If we don't explicitly declared it as a dependency in polyExplorer but need it in polyExplorer, can we declare it there?

Reply:
The problem is that it's not easy to handle updating of indirect dependencies, except via auditing. By doing this, we pin a specific version and also unify the versions used in the two places. But I understand the general principle, do not hoist if it's not used in strictly more than two places. Also: conflicts. So I think I'd better close it now.